adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,967 @@
+
+template <typename KernelType>
+std::optional<event> launch(const queue& q, range<1> r, const KernelType& k);                                         (4)
+
+template <typename KernelType>
+std::optional<event> launch(const queue& q, range<2> r, const KernelType& k);                                         (5)
+
+template <typename KernelType>
+std::optional<event> launch(const queue& q, range<3> r, const KernelType& k);                                         (6)
+
+}
+----
+
+_Effects_ (1-3): Equivalent to: +
+[source,sycl]
+----
+event ev = q.submit([&](handler& h) {
+  reqs.register-events(h);
+  reqs.register-accessors(h);
+  reqs.register-kernel-bundle(h);
+  h.parallel_for(r, k);
+});
+return (reqs.has-tracking()) ? ev : std::nullopt;
+----
+
+_Effects_ (4-6): Equivalent to: [code]#return launch(q, r, {}, k);#.
+
+'''
+
+// Launch a basic parallel_for with a sycl::kernel object.
+// New form of handler::parallel_for(range, ...) without set_args.
+.[apititle]#launch# (kernel object)
+[source,role=synopsis,id=api:launch-kernel]
+----
+namespace sycl::khr {
+
+template <typename... Args, typename... Requirements>
+std::optional<event> launch(const queue& q, range<1> r, const requirements<Requirements>& reqs, const kernel& k, Args&&... args); (1)
+
+template <typename... Args, typename... Requirements>
+std::optional<event> launch(const queue& q, range<2> r, const requirements<Requirements>& reqs, const kernel& k, Args&&... args); (2)
+
+template <typename... Args, typename... Requirements>
+std::optional<event> launch(const queue& q, range<3> r, const requirements<Requirements>& reqs, const kernel& k, Args&&... args); (3)
+
+template <typename... Args>
+std::optional<event> launch(const queue& q, range<1> r, const kernel& k, Args&&... args);                                         (4)
+
+template <typename... Args>
+std::optional<event> launch(const queue& q, range<2> r, const kernel& k, Args&&... args);                                         (5)
+
+template <typename... Args>
+std::optional<event> launch(const queue& q, range<3> r, const kernel& k, Args&&... args);                                         (6)
+
+}
+----
+
+_Constraints_ (1-3): [code]#Requirements# does not contain a
+[code]#kernel_bundle#.
+
+_Effects_ (1-3): Equivalent to: +
+[source,sycl]
+----
+event ev = q.submit([&](handler& h) {
+  reqs.register-events(h);
+  reqs.register-accessors(h);
+  h.set_args(args...);
+  h.parallel_for(r, k);
+});
+return (reqs.has-tracking()) ? ev : std::nullopt;
+----
+
+_Effects_ (4-6): Equivalent to: [code]#+return launch(q, r, {}, k, args...);+#.
+
+'''
+
+// Launch a basic parallel_for with a function object and reductions.
+// New form of parallel_for(range, reduction, ...)
+.[apititle]#launch_reduce# (kernel function)
+[source,role=synopsis,id=api:launch_reduce]
+----
+namespace sycl::khr {
+
+template <typename KernelType, typename... Requirements, typename... Reductions>
```

Comment:
This is a minor point, but I see that for these overloads we have `Requirements` and then `Reductions`, but in the overloads above for taking a kernel object, we have `Args` and then `Requirements`, should we be consistent in having the variadic function arguments as the last template arg pack? Or is there a reason it's ordered in this way.

Reply:
There's no reason things are ordered this way.  I noticed the inconsistency myself, but figured there was a good chance we'd be reordering the arguments anyway and so things would have to change.

I wanted to collect some feedback about which order things should appear in.  Should the template parameters always be in the same order across different overloads (which would put `Requirements` at the end)?  Or should the template parameters always match the order of the function arguments?  Or something else?